File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/java/io/airbyte/commons/functional/CheckedConsumer.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.functional;

import org.apache.commons.lang3.function.FailableConsumer;

@FunctionalInterface
public interface CheckedConsumer<T, E extends Throwable> extends FailableConsumer<T, E> {

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/functional/CheckedConsumer.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.commons.functional

import org.apache.commons.lang3.function.FailableConsumer

fun interface CheckedConsumer<T, E : Throwable?> : FailableConsumer<T, E>
